src/gwproxy/gwproxy.c

```diff
@@ -992,6 +1065,12 @@ static int gwp_load_s5auth_add_user(struct gwp_socks5_auth *s5a,
 		return -EINVAL;
 	}
 
+	/* Check for empty username */
+	if (!*u) {
+		free(u);
+		return -EINVAL;
+	}
+
 	s5a->users[s5a->nr].u = u;
```

Comment:
This is not needed because the check `strlen(u) > 255` above has guaranteed `!*u` is false.